cpp/src/io/json/read_json.cu

```diff
@@ -219,6 +229,18 @@ table_with_metadata read_json(host_span<std::unique_ptr<datasource>> sources,
 
   auto const buffer = get_record_range_raw_input(sources, reader_opts, stream);
 
+  // If input JSON buffer has single quotes and option to normalize single quotes is enabled,
+  // invoke pre-processing FST
+  if (reader_opts.is_enabled_normalize_single_quotes()) {
+    auto buffer_span = cudf::device_span<std::byte const>(
+      reinterpret_cast<const std::byte*>(buffer.data()), buffer.size());
+    return device_parse_nested_json(cudf::io::json::detail::normalize_single_quotes(
+                                      buffer_span, stream, rmm::mr::get_current_device_resource()),
+                                    reader_opts,
+                                    stream,
+                                    mr);
+  }
```

Comment:
Couple of issues with this approach:
original `buffer` is still allocated during `device_parse_nested_json`, even though it's not used after `normalize_single_quotes`.
We have duplicated call to `device_parse_nested_json`

I think we can fix both in one fell swoop:
```suggestion
  if (reader_opts.is_enabled_normalize_single_quotes()) {
    buffer = normalize_single_quotes(buffer, stream, rmm::mr::get_current_device_resource());
  }
```

Reply:
Thank you for the explanation, fixed with the modified `normalize_single_quotes` declaration.